tests/core/providers/test_websocket_provider.py

```diff
@@ -64,6 +60,13 @@ def test_websocket_provider_timeout(w3):
         w3.eth.accounts
 
 
+@patch('web3.providers.websocket.WebsocketProvider.coro_make_request')
+def test_websocket_provider_future_timeout(coro_make_request_mock, w3):
+    coro_make_request_mock.side_effect = asyncio.coroutine(lambda o: asyncio.sleep(0.02))
```

Comment:
I guess what this does is that the request doesn't actually go to the WebSocket server. So this is trying to simulate the behavior where the coroutine that makes a request to the WebSocket server gets stuck for some reason and this can only happen when the WebSocket library doesn't respect the `websocket_timeout `  and keeps waiting for some reason?
https://github.com/ethereum/web3.py/blob/0aa00c8e44ee0be116c8ad2baea952c05ff7ad34/web3/providers/websocket.py#L98-L109
This could be a bug in the WebSocket library and I don't know if anyone has a test case to reproduce it and report it upstream. I'm not opposed to this change, but I'm not fully convinced by what is causing this!

PS: I'm not the maintainer of this project. However, the WebSocket code was initially added by me and I try to stay by its side.

Reply:
Yes, pretty much. This simulates the situation where we get a response in a timely manner from the websocket server, but the future response takes too long to resolve (i.e. takes longer than self.websocket_timeout to resolve).

Can you help me tag the right person to get this PR merged? Otherwise, I can bug you guys later on gitter. Thanks

Reply:
@gohuygo I will take a look soon! @voith is more familiar with the websocket provider than I am so I was waiting for his questions to get resolved before I jumped in.

Reply:
> Can you help me tag the right person to get this PR merged?

That'd be @kclowes!

> I was waiting for his questions to get resolved before I jumped in

My only concern here is not knowing what is causing the WebSocket library to not respect the  `websocket_timeout`. As you can see in the testcase below for `websocket_timeout` works fine when a  request gets timed out. 
https://github.com/ethereum/web3.py/blob/0aa00c8e44ee0be116c8ad2baea952c05ff7ad34/tests/core/providers/test_websocket_provider.py#L62-L64

> takes longer than self.websocket_timeout to resolve

A possible problem I see with `future.result(self.websocket_timeout)` is that if there's a large response that is being streamed, it can get canceled if it exceeds the timeout. This can happen with filters when you request a big chunk of data. Thinking about this now, I'm 👎 on this change.

Instead of mocking what I really wanted to see is the case when the websocket request keeps hanging and doesn't respect the `websocket_timeout`.  Since PR initially started out with a fix for https://github.com/ethereum/web3.py/issues/1449, I was looking for a test case that would simulate that behavior.

TLDR; I'm 👍 if someone proves to me that the websocket library is not respecting `websocket_timeout`.Ideally, I would still like to see this fixed upstream.
I'm 👎 as this can cancel a streaming response.

Reply:
Good point. Let's close this PR then. Thanks for the review